src/test/java/org/elasticsearch/search/highlight/HighlighterSearchTests.java

```diff
@@ -659,24 +659,36 @@ public void testPlainHighlighterDocumentAnalyzer() throws Exception {
                     .field("type", "string")
                     .field("index", "not_analyzed")
                 .endObject()
-                .startObject("text")
+                .startObject("field1")
                     .field("type", "string")
                 .endObject()
+                .startObject("field2")
+                    .field("type", "string")
+                    .field("analyzer", "standard")
+                .endObject()
             .endObject()
             .endObject().endObject()).execute().actionGet();
         ensureYellow();
 
         index("test", "type1", "1",
                 "language_analyzer", "english",
-                "text", "Look at me, I'm eating cars.");
+                "field1", "Look at me, I'm eating cars.",
+                "field2", "Look at me, I'm eating cars.");
         refresh();
 
         SearchResponse response = client().prepareSearch("test")
-                .setQuery(QueryBuilders.matchQuery("text", "car"))
+                .setQuery(QueryBuilders.matchQuery("field1", "car"))
+                .addHighlightedField(
+                        new HighlightBuilder.Field("field1").preTags("<1>").postTags("</1>").requireFieldMatch(true))
+                .get();
+        assertHighlight(response, 0, "field1", 0, 1, equalTo("Look at me, I'm eating <1>cars</1>."));
+
+        response = client().prepareSearch("test")
+                .setQuery(QueryBuilders.matchQuery("field2", "cars"))
                 .addHighlightedField(
-                        new HighlightBuilder.Field("text").preTags("<1>").postTags("</1>").requireFieldMatch(true))
+                        new HighlightBuilder.Field("field2").preTags("<1>").postTags("</1>").requireFieldMatch(true))
                 .get();
-        assertHighlight(response, 0, "text", 0, 1, equalTo("Look at me, I'm eating <1>cars</1>."));
+        assertHighlight(response, 0, "field2", 0, 1, equalTo("Look at me, I'm eating <1>cars</1>."));
```

Comment:
Yes, it fails.

Since field2 uses standard analyzer, `cars` is analyzed as `cars`. Without this fix, query string "cars" is analyzed by standard analyzer (set on the field) and field's value "Look at me, I'm eating cars." is analyzed by "language_analyzer" (specified by `_analzyer`) during highlighting. The former becomes `cars` and the latter becomes `car`. So, highlight doesn't work.

Yes, this test is a bit confusing... but it may be because of it's nature. We see this issue when `_analyzer` path is set AND analyzer is set on the field. But you are right. better to write separate test for it. Will push another commit.